lib/Adafruit_HMC5883_Unified/Adafruit_HMC5883_U.cpp

```diff
@@ -0,0 +1,266 @@
+}
+
+/**************************************************************************/
+/*!
+    @brief  Abstract away platform differences in Arduino wire library
+*/
+/**************************************************************************/
+byte Adafruit_HMC5883_Unified::read8(byte address, byte reg) {
+  byte value;
+
+  Wire.beginTransmission(address);
+#if ARDUINO >= 100
+  Wire.write((uint8_t)reg);
+#else
+  Wire.send(reg);
+#endif
+  Wire.endTransmission();
+  Wire.requestFrom(address, (byte)1);
+#if ARDUINO >= 100
+  value = Wire.read();
+#else
+  value = Wire.receive();
+#endif
+  Wire.endTransmission();
+
+  return value;
+}
+
+/**************************************************************************/
+/*!
+    @brief  Reads the raw data from the sensor
+*/
+/**************************************************************************/
+void Adafruit_HMC5883_Unified::read() {
+  // Read the magnetometer
+  Wire.beginTransmission((byte)HMC5883_ADDRESS_MAG);
+#if ARDUINO >= 100
+  Wire.write(HMC5883_REGISTER_MAG_OUT_X_H_M);
+#else
+  Wire.send(HMC5883_REGISTER_MAG_OUT_X_H_M);
+#endif
+  Wire.endTransmission();
+  Wire.requestFrom((byte)HMC5883_ADDRESS_MAG, (byte)6);
+
+  // Wait around until enough data is available
+  while (Wire.available() < 6)
+    ;
+
+// Note high before low (different than accel)
+#if ARDUINO >= 100
+  uint8_t xhi = Wire.read();
+  uint8_t xlo = Wire.read();
+  uint8_t zhi = Wire.read();
+  uint8_t zlo = Wire.read();
+  uint8_t yhi = Wire.read();
+  uint8_t ylo = Wire.read();
+#else
+  uint8_t xhi = Wire.receive();
+  uint8_t xlo = Wire.receive();
+  uint8_t zhi = Wire.receive();
+  uint8_t zlo = Wire.receive();
+  uint8_t yhi = Wire.receive();
+  uint8_t ylo = Wire.receive();
+#endif
+
+  // Shift values to create properly formed integer (low byte first)
+  _magData.x = (int16_t)(xlo | ((int16_t)xhi << 8));
+  _magData.y = (int16_t)(ylo | ((int16_t)yhi << 8));
+  _magData.z = (int16_t)(zlo | ((int16_t)zhi << 8));
+
+  // ToDo: Calculate orientation
+  _magData.orientation = 0.0;
+}
+
+/***************************************************************************
+ CONSTRUCTOR
+ ***************************************************************************/
+
+/**************************************************************************/
+/*!
+    @brief  Instantiates a new Adafruit_HMC5883 class
+*/
+/**************************************************************************/
+Adafruit_HMC5883_Unified::Adafruit_HMC5883_Unified(int32_t sensorID) {
+  _sensorID = sensorID;
+}
+
+/***************************************************************************
+ PUBLIC FUNCTIONS
+ ***************************************************************************/
+
+/**************************************************************************/
+/*!
+    @brief  Setups the HW
+*/
+/**************************************************************************/
+bool Adafruit_HMC5883_Unified::begin() {
+  // Enable I2C
+  Wire.begin();
```

Comment:
This is not needed/undesired, as ESPEasy does the initialization of the I2C bus already.

Reply:
I just used the original lib. Shall I change it?

Reply:
Please comment it out.